admin-dev/themes/new-theme/js/pages/order/view/order-view-page.js

```diff
@@ -337,4 +308,68 @@ export default class OrderViewPage {
   getActivePage() {
     return $(OrderViewPageMap.productsTablePagination).find('.active span').get(0);
   }
+
+  refreshProductsList(orderId) {
+    const $loadingSpinner = $(OrderViewPageMap.productsPanel).find('.spinner-order-products-container#orderProductsLoading');
+    $loadingSpinner.show();
+
+    const $tablePagination = $(OrderViewPageMap.productsTablePagination);
+    const numRowsPerPage = $tablePagination.data('numPerPage');
+    const initialNumProducts = $(OrderViewPageMap.productsTableRows).length;
+
+    $.ajax(this.router.generate('admin_orders_get_products', {orderId}))
+        .done((response) => {
+          // Delete previous product lines
+          $(OrderViewPageMap.productsTable).find(OrderViewPageMap.productsTableRows).remove();
+
+          $(OrderViewPageMap.productsTable + ' tbody').prepend(response);
+
+          $loadingSpinner.hide();
+
+          const newNumProducts = $(OrderViewPageMap.productsTableRows).length;
+          const initialPagesNum = Math.max(Math.ceil(initialNumProducts / numRowsPerPage), 1);
+          const newPagesNum = Math.ceil(newNumProducts / numRowsPerPage);
+
+          this.orderProductRenderer.updateNumProducts(newNumProducts);
+
+          // Update pagination
+          if (newPagesNum > initialPagesNum) {
+            this.orderProductRenderer.paginationAddPage(newPagesNum);
```

Comment:
Be careful this won't work when you remove a product if the number of pages decreased It would be easier to actually just set the number of pages with a new method and get rid of these unstable addPage/removePage methods
All this should be handled by a single initPagination method that is used on page loading as well